File: prob_36.py
# 88 leetcode
# time - O(N), space - O(1)
# start from last index of both arrays and compare and add nums2 to nums1 - sorting inplace.
class Solution:
    def merge(self, nums1, m, nums2, n) :
        """
        Do not return anything, modify nums1 in-place instead.
        """
        #
        while m > 0 and n > 0:
            if nums1[m-1] < nums2[n-1]:
                nums1[m+n-1]=nums2[n-1]
                n = n-1
            else:
                nums1[m+n-1]=nums1[m-1]
                m = m-1
        if m == 0:
            while n > 0:
                print(nums1[n-1])
                nums1[n-1] = nums2[n-1]
                n = n-1

File: prob36.java
// 30 Big N Problem #36 {Easy}
// 88. Merge Sorted Array
// Time Complexity :O(m+n)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :Yes
// Any problem you faced while coding this :None


// Your code here along with comments explaining your approach
//The key to solve this problem is moving element of nums1 and nums2 backwards. If nums2 has some elements left after nums1 is done, also need to handle that case.
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        while(m>0 && n>0){
           if(nums1[m-1]>nums2[n-1]){
               nums1[m+n-1]=nums1[m-1];
               m--;
           }else{
               nums1[m+n-1]=nums2[n-1];
               n--;
           }
       }
        
        while(n>0){
            nums1[m+n-1]=nums2[n-1];
            n--;
        }
    }
}
